MinecraftClient/ChatBots/MineCube.cs

```diff
@@ -0,0 +1,517 @@
+            int[] iterateY = getNumbersFromTo(0, Convert.ToInt32(Math.Round(vectorToStopPosition.Y))).ToArray();
+            int[] iterateZ = getNumbersFromTo(0, Convert.ToInt32(Math.Round(vectorToStopPosition.Z))).ToArray();
+
+            LogDebugToConsole("Iterate on X: 0-" + (iterateX.Length - 1).ToString() + " Y: 0-" + (iterateY.Length - 1).ToString() + " Z: 0-" + (iterateZ.Length - 1).ToString());
+
+            // Iterate through all coordinates relative to the start block
+            foreach (int y in iterateY)
+            {
+                Layer tempLayer = new Layer();
+                foreach (int x in iterateX)
+                {
+                    Row tempRow = new Row();
+                    foreach (int z in iterateZ)
+                    {
+                        Location tempLocation = new Location(Math.Round(startBlock.X + x), Math.Round(startBlock.Y + y), Math.Round(startBlock.Z + z));
+                        if (isMinable(GetWorld().GetBlock(tempLocation).Type))
+                        {
+                            tempRow.BlocksToMine.Add(tempLocation);
+                        }
+                    }
+                    if (tempRow.BlocksToMine.Count > 0)
+                    {
+                        tempLayer.addRow(tempRow);
+                    }
+                }
+                if (tempLayer.RowsToMine.Count > 0)
+                {
+                    cubeToMine.addLayer(tempLayer);
+                }
+            }
+
+            // Remove later ;D
+            printCubeToConsole(cubeToMine);
+
+            if (Settings.DebugMessages)
+            {
+                printCubeToConsole(cubeToMine);
+            }
+
+            return cubeToMine;
+        }
+
+        /// <summary>
+        /// Get all numbers between from and to.
+        /// </summary>
+        /// <param name="start">Number to start</param>
+        /// <param name="end">Number to stop</param>
+        /// <returns>All numbers between the first, including the stop number</returns>
+        public List<int> getNumbersFromTo(int start, int stop)
+        {
+            List<int> tempList = new List<int>();
+            if (start <= stop)
+            {
+                for (int i = start; i <= stop; i++)
+                {
+                    tempList.Add(i);
+                }
+            }
+            else
+            {
+                for (int i = start; i >= stop; i--)
+                {
+                    tempList.Add(i);
+                }
+            }
+            return tempList;
+        }
+
+        public Func<Location, Location> getHeadLocation = locFeet => new Location(locFeet.X, locFeet.Y + 1, locFeet.Z);
+
+        /// <summary>
+        /// Checks whether a material is minable
+        /// </summary>
+        /// <param name="block">Block that should be checked</param>
+        /// <returns>Is block minable</returns>
+        private bool isMinable(Material block)
```

Comment:
```suggestion
        private bool IsMinable(Material block)
```

***

You may want to move this method to [MaterialExtensions.cs](https://github.com/ORelio/Minecraft-Console-Client/blob/master/MinecraftClient/Mapping/MaterialExtensions.cs)